### PR TITLE
fix(openfga): allow service and deployment annotations

### PR DESCRIPTION
**Pull request by PedroArvela**
## Description

Specify `.annotations` and `.service.annotations` in the chart, that allow adding additional annotations to the deployment and to the service resources, respectively.

## References

#35 

## Review Checklist
- [ ] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [x] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [x] The correct base branch is being used, if not `main`
- [x] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

**Comment by jon-whit** [comment]
@PedroArvela for documentation purposes, please add the new annotations to the `values.schema.json` file as well.

**Comment by PedroArvela** [comment]
Added to the json schema!

File: charts/openfga/values.schema.json [hunk 1]
```diff
@@ -313,6 +313,40 @@
         "allowEvaluating1_0Models": {
             "type": ["boolean", "null"],
             "description": "allow calling Check, Expand, ListObjects, Write Assertions with models that have 1.0 schema version"
+        },
+        "annotations": {
```
Comment (jon-whit) [inline charts/openfga/values.schema.json:317]: I think there may have been a misunderstanding here.

If we added the `annotations` to the top-level `values.yaml` body, then we only need an entry in the `values.schema.json` file for that single key/value pair. We don't need to enumerage all of the service, deployment, etc.. manifestations those annotations apply to.

We can remove the nested definitions for for `service.annotations` and the others. All we need is the `annotations` entry here.

**Review by jon-whit — CHANGES_REQUESTED** [review]
Left a comment regarding the annotations entry in the `values.schema.json` file.

File: charts/openfga/values.schema.json (hunk 1 above)
Comment (PedroArvela) [reply charts/openfga/values.schema.json:317]: @jon-whit let know what you think based on my reply above, and based on it I'll refactor the PR to match the decision :)

Comment (jon-whit) [reply charts/openfga/values.schema.json:317]: @PedroArvela I'll take a look today. Sorry for the delay.

**Review by jon-whit — APPROVED** [review]

**Pull request merged**